### PR TITLE
Improve BankID login design

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Expose BankID login status through API
- Reflect every BankId status in UI
- Apply designs

## Justify why they are needed

Intermediate step, to be followed by more improvements

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1525** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1525" target="_blank">Graphite</a> 
    * **PR #1529** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1529" target="_blank">Graphite</a> 
      * **PR #1549** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1549" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1549?utm_source=stack-comment).

File: apps/store/src/services/bankId.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  Error = 'Error',
+}
+
+type BankIdLoginOptions = {
+  shopSessionId: string
+  ssn: string
+  onCompleted: () => void
+}
+export const useBankIdLogin = ({ shopSessionId, ssn, onCompleted }: BankIdLoginOptions) => {
+  const [state, setState] = useState(BankIdState.Idle)
```
Comment (robinandeer) [inline apps/store/src/services/bankId.ts:22]: Isn't there a case where we have an ongoing bank ID session after user e.g. reloads the page unexpectedly?

File: apps/store/src/services/bankId.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      const accessToken = await exchangeAuthorizationCode(authorizationCode)
+      saveAccessToken(accessToken)
+      await authenticateShopSession()
+      onCompleted()
+    } catch (error) {
+      datadogLogs.logger.warn('Failed to authenticate', { error })
+      setState(BankIdState.Error)
+    }
+  }, [authenticateShopSession, onCompleted, ssn])
+  return [state, startLogin] as const
```
Comment (robinandeer) [inline apps/store/src/services/bankId.ts:50]: nit: reverse order to match apollo client API

File: apps/store/src/services/bankId.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+type BankIdLoginOptions = {
+  shopSessionId: string
+  ssn: string
+  onCompleted: () => void
+}
+export const useBankIdLogin = ({ shopSessionId, ssn, onCompleted }: BankIdLoginOptions) => {
+  const [state, setState] = useState(BankIdState.Idle)
+  const [authenticateShopSession] = useShopSessionAuthenticateMutation({
```
Comment (robinandeer) [inline apps/store/src/services/bankId.ts:23]: nit: add comment about exposing error

File: apps/store/src/components/PriceCalculator/SsnSeSection.tsx [hunk 4]
```diff
@@ -1,14 +1,17 @@
 import { datadogLogs } from '@datadog/browser-logs'
 import styled from '@emotion/styled'
 import { useTranslation } from 'next-i18next'
-import { FormEventHandler } from 'react'
-import { Button, Space, Text, theme } from 'ui'
-import { BankIdLogin } from '@/components/BankIdLogin'
+import { FormEventHandler, ReactElement } from 'react'
+import { WarningTriangleIcon } from 'ui/src/icons/WarningTriangleIcon'
```
Comment (robinandeer) [inline apps/store/src/components/PriceCalculator/SsnSeSection.tsx:5]: update import -- should be exported/imported from `ui`

File: apps/store/src/components/PriceCalculator/SsnSeSection.tsx [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
+import { BankIdIcon, Button, Space, Text, theme } from 'ui'
+import { BankIdLoginForm } from '@/components/BankIdLogin'
+import { TickIcon } from '@/components/CheckoutPage/TickIcon'
```
Comment (robinandeer) [inline apps/store/src/components/PriceCalculator/SsnSeSection.tsx:8]: nit: move to global component or make part of `ui` package

**Review by robinandeer — APPROVED** [review]
Added via GiphyFantastic work! Left a bunch of comments but I think the code itself is good to go :D

**Comment by alebedev** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: LNt7OB37uAsddXEUWHM6.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1549)

**Pull request merged**